lib/bamboo/mailer.ex

```diff
@@ -62,19 +62,14 @@ defmodule Bamboo.Mailer do
 
   defmacro __using__(opts) do
     quote bind_quoted: [opts: opts] do
-      config = Bamboo.Mailer.parse_opts(__MODULE__, opts)
-
-      @adapter config.adapter
-      @config config
-
       @spec deliver_now(Bamboo.Email.t) :: Bamboo.Email.t
       def deliver_now(email) do
-        Bamboo.Mailer.deliver_now(@adapter, email, @config)
+        Bamboo.Mailer.deliver_now(adapter, email, config)
```

Comment:
Random thoughts: This parses the config 2x per delivery, once to get the config, but again to get adapter which just calls `config.adapter`. Maybe something like:

```
conf = config
Bamboo.Mailer.deliver_now(conf.adapter, email, conf)
```

Premature optimization. :)